Failed tests (11597): TestBootstrapNewStore

The following test appears to have failed:

[#11597](https://circleci.com/gh/cockroachdb/cockroach/11597):

```
I0119 19:54:22.181776 437 server/node.go:377  bootstrapped store store=1:3 ([]=)
I0119 19:54:22.287233 437 gossip/client.go:87  closing client to 127.0.0.1:0: gossip/client.go:78: client closed
I0119 19:54:22.288005 437 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
I0119 19:54:22.288165 437 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
I0119 19:54:22.288260 437 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- FAIL: TestBootstrapNewStore (1.18s)
	node_test.go:186: server/node_test.go:185: condition failed to evaluate true within 1s
=== RUN   TestNodeJoin
I0119 19:54:22.291715 437 raft/raft.go:446  [group 1] 1 became follower at term 5
I0119 19:54:22.291880 437 raft/raft.go:234  [group 1] newRaft 1 [peers: [1], term: 5, commit: 10, applied: 10, lastindex: 10, lastterm: 5]
I0119 19:54:22.291948 437 raft/raft.go:526  [group 1] 1 is starting a new election at term 5
I0119 19:54:22.292025 437 raft/raft.go:459  [group 1] 1 became candidate at term 6
I0119 19:54:22.292085 437 raft/raft.go:508  [group 1] 1 received vote from 1 at term 6
I0119 19:54:22.292169 437 raft/raft.go:487  [group 1] 1 became leader at term 6
I0119 19:54:22.293993 437 storage/replica_command.go:1214  range 1: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC +1453233263.293s
I0119 19:54:22.308623 437 raft/raft.go:446  [group 1] 1 became follower at term 6
--
I0119 19:54:43.645108 437 client/db.go:485  failed batch: failed to send RPC: too many errors encountered (1 of 1 total): server/node.go:494: node 1 stopped
W0119 19:54:43.645310 437 ts/db.go:96  error writing time series data: failed to send RPC: too many errors encountered (1 of 1 total): server/node.go:494: node 1 stopped
I0119 19:54:43.646833 437 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestMetricsRecording (1.42s)
	<autogenerated>:28: prior leak detected, leaktest disabled
FAIL
FAIL	github.com/cockroachdb/cockroach/server	32.812s
=== RUN   TestNodeStatusMonitor
--- PASS: TestNodeStatusMonitor (0.01s)
=== RUN   TestNodeStatusRecorder
--- PASS: TestNodeStatusRecorder (0.04s)
=== RUN   TestRuntimeStatRecorder
I0119 19:54:32.982961 472 server/status/runtime.go:137  runtime stats: 11 goroutines, 4.90MiB active, 50000000.00cgo/sec, 546430.00/390300.00 %(u/s)time, 30363.96 %gc (2x)
--- PASS: TestRuntimeStatRecorder (0.01s)
=== RUN   TestNodeEventFeed
--- PASS: TestNodeEventFeed (0.01s)
=== RUN   TestServerNodeEventFeed

```
Please assign, take a look and update the issue accordingly.

--- a/server/node_test.go
+++ b/server/node_test.go
@@ -43,6 +43,8 @@ import (
 	"github.com/cockroachdb/cockroach/util/stop"
 )
 
+const testTimeout = 3 * time.Second
+
 // createTestNode creates an rpc server using the specified address,
 // gossip instance, KV database and a node using the specified slice
 // of engines. The server, clock and node are returned. If gossipBS is
@@ -67,7 +69,7 @@ func createTestNode(addr net.Addr, engines []engine.Engine, gossipBS net.Addr, t
 	g := gossip.New(nodeRPCContext, testContext.GossipBootstrapResolvers)
 	if gossipBS != nil {
 		// Handle possibility of a :0 port specification.
-		if gossipBS == addr {
+		if gossipBS.Network() == addr.Network() && gossipBS.String() == addr.String() {
 			gossipBS = ln.Addr()
 		}
 		r, err := resolver.NewResolverFromAddress(gossipBS)
@@ -80,11 +82,12 @@ func createTestNode(addr net.Addr, engines []engine.Engine, gossipBS net.Addr, t
 	ctx.Gossip = g
 	retryOpts := kv.GetDefaultDistSenderRetryOptions()
 	retryOpts.Closer = stopper.ShouldDrain()
-	sender := kv.NewDistSender(&kv.DistSenderContext{
+	distSender := kv.NewDistSender(&kv.DistSenderContext{
 		Clock:           ctx.Clock,
 		RPCContext:      nodeRPCContext,
 		RPCRetryOptions: &retryOpts,
 	}, g)
+	sender := kv.NewTxnCoordSender(distSender, ctx.Clock, false, nil, stopper)
 	ctx.DB = client.NewDB(sender)
 	// TODO(bdarnell): arrange to have the transport closed.
 	// (or attach LocalRPCTransport.Close to the stopper)
@@ -186,11 +189,14 @@ func TestBootstrapNewStore(t *testing.T) {
 
 	// Non-initialized stores (in this case the new in-memory-based
 	// store) will be bootstrapped by the node upon start. This happens
-	// in a goroutine, so we'll have to wait a bit (maximum 1s) until
-	// we can find the new node.
-	if err := util.IsTrueWithin(func() bool { return node.stores.GetStoreCount() == 3 }, 1*time.Second); err != nil {
-		t.Error(err)
-	}
+	// in a goroutine, so we'll have to wait a bit until we can find the
+	// new node.
+	util.SucceedsWithin(t, testTimeout, func() error {
+		if n := node.stores.GetStoreCount(); n != 3 {
+			return util.Errorf("expected 3 stores but got %d", n)
+		}
+		return nil
+	})
 
 	// Check whether all stores are started properly.
 	if err := node.stores.VisitStores(func(s *storage.Store) error {
@@ -229,7 +235,7 @@ func TestNodeJoin(t *testing.T) {
 	defer stopper2.Stop()
 
 	// Verify new node is able to bootstrap its store.
-	if err := util.IsTrueWithin(func() bool { return node2.stores.GetStoreCount() == 1 }, time.Second); err != nil {
+	if err := util.IsTrueWithin(func() bool { return node2.stores.GetStoreCount() == 1 }, testTimeout); err != nil {
 		t.Fatal(err)
 	}
 
@@ -452,7 +458,7 @@ func TestStatusSummaries(t *testing.T) {
 
 	// Wait for full replication of initial ranges.
 	initialRanges := int32(ExpectedInitialRangeCount())
-	util.SucceedsWithin(t, time.Second, func() error {
+	util.SucceedsWithin(t, testTimeout, func() error {
 		for i := 1; i <= int(initialRanges); i++ {
 			if s.RaftStatus(roachpb.RangeID(i)) == nil {
 				return util.Errorf("Store %d replica %d is not present in raft", s.StoreID(), i)

